Polish classic product editor CSS

tomalec opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR introduces a UI polish to the classic product and coupon editors. As proposed in pfrEX1-mz-p2.
The intent is to CSS-only change that modernizes the UI without disrupting the merchants and 3PDs flows.

I Added automattic/color-studio dependency, and used its colors. However, I didn't change existing rules to use that pallete.

<!-- Begin testing instructions -->

#### Before
![image](https://github.com/user-attachments/assets/84901bad-b010-436f-b985-b9414966439f)

#### After
![image](https://github.com/user-attachments/assets/15b78169-9e39-4e04-944a-4e7c22a887b6)


### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the Figma design from pfrEX1-mz-p2 to compare
2. Install a few canonical extensions (p6q6po-e59-p2), like AutomateWoo, Google for WooCommerce, WooCommerce Subscriptions, Product Bundles, etc.
3. Open the updated pages:
	1.  "Add product": `Products > Add new product`
	2. "Edit product": `Products > All products > (pick one) > edit`
	3. "Add coupon": `Marketing > Coupons >Add new Coupon`
	4. "Edit coupon": `Marketing > Coupons >(pick one) > edit`
4. Browse through the view looking for mismatches, switch tabs in editor view, product/coupon-data box, collapse meta boxes, add/remove images, etc.
5. Resize the window to smaller one, to check the responsiveness
6. Disable JavaScript (open browser dev tools (F12) >  `ctrl+shift+p` > type "javascript"), reload the page, test the page in no-js mode.

<!-- End testing instructions -->

### Additional notes

#### Differences to the Figma design:

1. Whenever the "Blueberry" `#3858E9` color is used, I used `var(--wp-admin-theme-color, $studio-wordpress-blue);`. Which means:
	- The Blueberry color is defined in automattic/color-studio package **at version 4.0.0** as `$studio-wordpress-blue`
	- The Blueberry will be used there if no `--wp-admin-theme-color` defined
	- It will use `--wp-admin-theme-color` if there is one to keep the look consistent with other parts of the WordPress screen, and make native theming easier.
	- It makes the implementation consistent with other uses of **hard-coded** `#3858E9` in Woo monorepo.
	- :warning: Some other packages in the monorepo uses **older version of color studio library: 2.5.0**
	- :warning: I decided to use the same - older - version of color-studio, as I believe it will be updated for the entire monorepo, with the planned efforts of branding update.
	- :warning: Currently, `--wp-admin-theme-color` is set and overwritten multiple times. And, even though it's initially set to `#3858E9`, then it's overwritten over and over by `#007cba`. I guess it's somehow related to Dependency Extraction Webpack Plugin loading style dependencies from wc-components, wp-components, etc. 
	- :warning: I leave links and button-links colored as `#3858E9` unchanged to avoid overspecificity of selectors and leave it up to a separate project to update all the Woo 2.0 colors (p7bje6-6wQ-p2)
	
   I'm leaving it like that to keep the same color as the rest of Woo, especially since the color change was not highlighted in the design notes. Also, I believe the color palette, will be revised and updated by a different project.
![image](https://github.com/user-attachments/assets/377feb93-f42a-4f51-b682-34304f7fd515)
![image](https://github.com/user-attachments/assets/592859a5-eaa0-4606-8e15-2221e039d55d)
4. (agreed in P2) Image dropzones are not part of this iteration
7. (agreed in P2) We keep the tab icons for the product-coupon data metabox.
8. (agreed in P2) Publish metabox, following changes were removed from this iteration as they would require HTML changes:
	1. The buttons remain where they were
	2. publish, visibility, etc. entries remain aligned as before - with "float: left" flow instead of vertically aligned columns.
9. (agreed in P2) The vertical padding in buttons is not set, not to disrupt buttons that may appear on the page in a form of notice boxes coming from other extensions not anticipating the new layout.
10. (agreed in P2) From fields in product-/coupon-data are not centered.
11. (agreed in P2) Product short description's fullscreen button is where it was before.
12. (discussed in P2) Duplicated top header stays where it was.
13. (discussed in P2) "+ Add new category" remained a text link, not a button.
14. (discussed in P2) Thumbnails area in image gallery metabox, has slightly smaller padding, to fit three images in a row.
15. The border of the Title input is the same as for other inputs `#8c8f94`, which is darker than the on in Figma - `#BBBCBF`. I believe they should be consistent between inputs, and the `#BBBCBF` was not an intentional change.  @j111q. I used `#BBBCBF` for for pane separators.

#### Implementation details
1. I decided to wrap most of the new styles with `post-php.post-type-product, .post-new-php.post-type-product, .post-php.post-type-shop_coupon, .post-new-php.post-type-shop_coupon`. I did so with the intention to limit the risk of disrupting other views and flows unintentionally. I'd like to make sure that in case there is the same UI element elsewhere in Woo or it's reused by another extension on some other page (like meta box, description editor, media button, etc.), we will not affect it there. This approach has a few downsides:
	- Leaves potentially dead code, if there are elements not used elsewhere
	- Increases the amount of bloated code, as instead of just removing the rule, we need to add more to overwrite it
	- Increases the specificity of selectors, which makes the rules harder to maintain, overwrite, and theme. 
   Because of those codes, I decided to style `#woocommerce-coupon-data, #woocommerce-product-data, #woocommerce-coupon-description` elements without those additional scope selectors. I believe product & coupon editor pages are the only views using those elements.
2. I made all the changes in the main `admin.scss` file and not separated it into a different file, to keep the files flow delivery intact, and as I believe we could eventually modernize other views and make those rules apply not only to product editor but globally.



### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

tomalec committed:
Adjust spacing in product data header

tomalec committed:
Adjust spacing and colors in product data tabs

tomalec committed:
Adjust option groups in product-data

tomalec committed:
Adjust all other postboxes for product editor

tomalec committed:
Adjust paddings for postboxes

tomalec committed:
Adjust spacing between postboxes

tomalec committed:
Fix collapsing

tomalec committed:
Adjust title field, Unify height of postbox headers

tomalec committed:
Share the styles with coupon editor as well

tomalec committed:
Adjust woocommerce-product-description

tomalec committed:
Adjust rich text tab buttons

tomalec committed:
Contain style update for product editor a bit more precise

tomalec committed:
Fix specificity for gap tweak in title input

tomalec committed:
Adjust button styles and wp-editor tools

tomalec committed:
Adjust help-tips in editor boxes

tomalec committed:
Bring back duplicated header to avoid layout shift

tomalec committed:
Adjust screen meta links

tomalec committed:
Adjust comments box

tomalec committed:
Adjust publish box

tomalec committed:
Change all links color to Woo Blueberry

tomalec committed:
Adjust categories metabox

tomalec committed:
Adjust custom fields

tomalec committed:
Adjust metabox header padding

tomalec committed:
Remove icons from publish box

tomalec committed:
Adjust tags list

tomalec committed:
Adjust tags list

tomalec committed:
Use Woo blueberry for button-links

tomalec committed:
Adjust image metabox

tomalec committed:
Adjust image metabox

tomalec committed:
Fix border-radius/background color for product-data

tomalec committed:
Adjust for no-js env

tomalec committed:
Adjust gaps/margins in metabox insides

tomalec committed:
Tweak inventory souldout form field

tomalec committed:
Tweak wrapping for media buttons

tomalec committed:
Tweaks for no-js mode & smaller screens

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

tomalec committed:
Use `--wp-admin-theme-color` instead of forced blueberry

to keep it consistent with other use in the codebase.
Use blueberry as a fallback value.

tomalec committed:
Move text-tertiary color definition to the top

tomalec committed:
Move gray-30 color definition to the top

tomalec committed:
Fix blueberry value to `3858E9`

tomalec committed:
Tweak borders for coupon-data

tomalec committed:
Isolate description editor changes to product and copoun pages only

tomalec committed:
Cleanup some vars

tomalec committed:
Remove dead code

tomalec committed:
Define separator color var

tomalec committed:
Tweak containment of wc-tabs styles

tomalec committed:
Bring back flex

accidentally removed in 4f9921d9da57c6101f94d4b35c7d0e6eddf996a9

tomalec committed:
Adjust E2E test case to the new tab name

tomalec committed:
Add changelog entry

tomalec commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -1196,8 +1215,7 @@ mark.amount {
 
-h2 .woocommerce-help-tip,
-.woocommerce-product-type-tip {
+h2 .woocommerce-help-tip {
```
`.woocommerce-product-type-tip` seems to be used only in `#woocommerce-product-data`

tomalec commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -8111,21 +8683,6 @@ table.bar_chart {
 	}
 }
 
-.wc-wp-version-gte-55 {
```
Minimum required WP version is higher.

tomalec committed:
Cater for AI suggested categories

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

tomalec committed:
Fix styles for screen-meta link buttons

tomalec committed:
Tweak paddings in screen meta links

tomalec committed:
Fix subcategory margins

tomalec committed:
Move .newtag rules close to each other

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

tomalec committed:
Add automattic/color-studio to client/legacy package

Use the studio colors

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -10,6 +10,24 @@
 @import "variables";
 @import "animation";
 @import "fonts";
+@import '@automattic/color-studio/dist/color-variables.scss';
```
```suggestion
@import "@automattic/color-studio/dist/color-variables";
```

to satisfy the WordPress coding standards:

```
13:9   ✖  Expected double quotes                   string-quotes
13:56  ✖  Unexpected extension ".scss" in @import  scss/at-import-partial-extension
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+	// Link color.
+	a,
+	.button-link  {
+	}
```
Is this empty block missing styling or did it miss being deleted?

tomalec replied:
Thanks, fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/3bc131af5df8d95500b00ca30bcc397447d67f29

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+		padding: 0;
+		margin: 0;
+	}
+	#titlediv #title {
+		margin: 0;
+	}
```
Looks like this block could be merged with the same selector above.

https://github.com/woocommerce/woocommerce/blob/a56700840aae885f8b78e4bd635be06e0397724c/plugins/woocommerce/client/legacy/css/admin.scss#L5677-L5679

![image](https://github.com/user-attachments/assets/0d4c81fc-f5a3-4723-941c-a377afdb7d3b)

tomalec replied:
That separation was on purpose.

I try to group rules based on what we - developers - are trying to achieve and conceptually style. Not, on what element is being styles - that's what browser will do either way.

```css
	// Title input.
	#titlediv {
		#title-prompt-text,
		#title  {
			padding: $gap-smallest $gap;
			// ...
			border-radius: $border-radius;
		}
	}
```

Styles the title input while

```css
	// Gap between Title and Slug edit should be 8px = $gap-smaller, so we need to clear margins.
	#poststuff #titlediv .inside {
		padding: 0;
		margin-top: $gap-smaller;
	}
	#edit-slug-box {
		padding: 0;
		margin: 0;
	}
	#titlediv #title {
		margin: 0;
	}
```

The margin rules cater to the spacing between the title and slug. In an ideal world, this would be just a gap property for `#titlediv`, but here, we need to overwrite margins set elsewhere.

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+			width: calc(100% - 2 * $gap);
+		}
+	}
+	//
```
Empty code comment.

tomalec replied:
fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/772870d4dbd50bb7fb6b831c8b851e502ef45ea6

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+		 * Sold individually option has help tip icon as a direct child, not as a child of form-field paragraph.
+		 * Let's make it look as it would be structured as others.
+		 */
+		 .options_group.inventory_sold_individually {
```
There is an extra space in the indentation.

```suggestion
		.options_group.inventory_sold_individually {
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+
+			// Active state.
+			&.tabs {
+			}
```
Are these empty blocks missing styling or did they miss being deleted?

tomalec replied:
Thanks, fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/3bc131af5df8d95500b00ca30bcc397447d67f29

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+		.tabs-panel {
+			padding: $gap;
+			margin: 0;
+			border:0;
```
Missing a space.

```suggestion
			border: 0;
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+			align-items: center;
+			gap: $gap;
+			flex: 1;
+			display: flex;
```
`display: flex;` is duplicated.

tomalec replied:
Removed in https://github.com/woocommerce/woocommerce/pull/52632/commits/f0d05bd2d90f1b91d3d499dba6b82c148f93bc7e

eason9487 reviewed (commented):
I haven't finished a full round yet, but I would like to send some feedback first.

Somehow `stylelint` doesn't seem to be running against the `plugins/woocommerce/client/legacy` directory any longer, and it's not even configured correctly to run. After I tried to get it to run on this PR change, there were about 945 linting errors.

![image](https://github.com/user-attachments/assets/5505c667-2802-4902-b686-cbad2c516415)

It's not easy to fix all of them, not to mention the above configure problems, but it's still suggested that some of them could be fixed first to satisfy the WordPress coding standards:

- `color-hex-case` expects lowercase
- `selector-pseudo-element-colon-notation` expects double colon
- `length-zero-no-unit` expects `0` instead of 0 with a unit, e.g. `0px`
- `function-parentheses-space-inside` expects no spaces after `(` and before `)`. For example:
   https://github.com/woocommerce/woocommerce/blob/a56700840aae885f8b78e4bd635be06e0397724c/plugins/woocommerce/client/legacy/css/admin.scss#L5646
- `selector-combinator-space-after` expects single space after `>`. For example:
   https://github.com/woocommerce/woocommerce/blob/a56700840aae885f8b78e4bd635be06e0397724c/plugins/woocommerce/client/legacy/css/admin.scss#L5805

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,528 @@ img.help_tip {
+
+	/* Style all buttons */
+	.button,
+	.wp-media-buttons .button,
```
Seems the priority of the `.button` selector in the previous line is high enough, and the `.wp-media-buttons .button` selector could be omitted?

tomalec replied:
It's a leftover from the time when I didn't wrap everything with `.post-php.post-type-product`.

Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/fadd9f6a1c8e29969fe68036fe658a6b69342b7e

tomalec committed:
Update coding standard for imports

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Remove redundant space

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Add missing space

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec commented:
> Are these empty blocks missing styling or did they miss being deleted?

This  makes me reconsider extracting those files into a separate file. I guess I may have lost something while merging/staging :( as the file is so big, my VSCode fails to load diffs there

tomalec committed:
Bring back accidentaly removed colors

This reverts part of
 "Add automattic/color-studio to client/legacy package"
 a56700840aae885f8b78e4bd635be06e0397724c.

 Address
 https://github.com/woocommerce/woocommerce/pull/52632/files#r1858179795
 https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1858196081

tomalec committed:
Use studio colors directly

tomalec committed:
Get gray-60 and white from color studio, lowercase hashes

tomalec committed:
stylelint: Remove whitespace from parenthasis

tomalec committed:
Stylelint: whitespaces 0unit

tomalec committed:
stylelint double colon

tomalec committed:
Remove empty comment

tomalec committed:
Bring back characters removed

in 836eb87f149130a5770dde2f592fb657adc782ad

tomalec committed:
move shared position rule for wc-tabs

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+
+			&::before {
+				font-size: inherit;
+				vertical-align: middle;
```
Maybe remove this style to make it align the text better?

The right one is after removal.
![image](https://github.com/user-attachments/assets/c28c3e4d-3ee4-4009-8088-29a2ee2a990e)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/5f9b0bf2c47ac2bce7aee3fcb2bba3ef4945e43a

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+		// Adjust spacing as the element is vertically aligned with flexbox, reset additional padding.
+		.show-settings::after {
+			padding: 0;
+			bottom: none;
```
`bottom: none;` doesn't seem to be a valid style. Maybe remove it as the `position` is set to static.

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/e8dba6ecc6e886f94e7be792f502e28de7a34357

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+			line-height: 46px;
+			height: auto;
+		}
+		#title-prompt-text {
```
It needs to be shifted down by 1px or increased line height by 2px to be vertically centered, and shifted right by 1px to have the same left spacing with the paired text input.

![image](https://github.com/user-attachments/assets/31a2f5cb-05d2-4f8a-bdd6-62a79cf4944e)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/afe8beab8b352da2c6045f1b97a3f6aff6ddf26d

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+		padding: 0;
+		margin-top: $gap-smaller;
+	}
+	#edit-slug-box {
```
The "Permalink" text color looks darker in the design document.

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/98fe28a3450c13e6fc880dec9d03466a7912bae5

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+
+	// Separate rule, not to overspecify the above ones with id selector.
+	#poststuff {
+		h2.postbox-header{
```
```suggestion
		h2.postbox-header {
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+			border-width: 0 0 2px 0;
+			border-bottom-color: transparent;
+			justify-content: center;
+			align-items: center;
```
Will the selected elements be set to flex container elsewhere? Otherwise these alignments won't take effect.

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/eaa45f8fce756c2c560ddfa82db2919bbb1cb953

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+		background: none;
+		padding: 0 $gap-smaller;
+		margin: 0;
+		display: inline-flex;
```
The display of `<a class="button">` in the "Get more options" tab is overridden to `inline-block`.

![image](https://github.com/user-attachments/assets/795cc7b7-fc13-4f0c-8d06-7a06a3cd4f7a)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/76e6e9cb5d2eb01f59ec386263c6afe6450d4ca8

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+			padding: 36px 48px;
+			display: flex;
+			flex-direction: column;
+			gap: $gap-small;
```
The vertical spacing between form fields is significantly larger than the design document.

![image](https://github.com/user-attachments/assets/3bf62e75-9c13-4fa0-b99f-bb833597a9b7)

tomalec replied:
I'll address this one after extracting code to a separate file, as I plan to remove the `flex` layout and get back to the previous padding-based implementation. I hope it will be less disruptive for extensions contributing their content like before.

tomalec replied:
Addressed in https://github.com/woocommerce/woocommerce/pull/52632/files/7360212ecdd57857565780bbcd660e734e7773ad..504c312c1549aa3f852923b5f28cd9e3a9df062f + https://github.com/woocommerce/woocommerce/pull/52632/commits/4bfc0b7a587e3846c554960ef1c6f7f7772bb7e4

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+
+			> a {
+				color: var(--Alias-text-text-tertiary, $alias-text-tertiary);
+				padding: $gap;
```
The height of category tabs seems to be smaller.

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/0841bb8a2c469bc0411aafa9873eaac04b94afd0

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+
+			// Active state.
+			&.tabs {
+				border-color:  var(--wp-admin-theme-color, $studio-wordpress-blue);
```
```suggestion
				border-color: var(--wp-admin-theme-color, $studio-wordpress-blue);
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+
+		> li {
+			border: 0;
+			margin: 0 0 -1px 0;
```
```suggestion
			margin: 0 0 -1px;
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+		.howto {
+			margin: 0;
+		}
+		.the-tagcloud {
+			margin-bottom: 0;
+		}
```
This selector appears to match nothing? The one I could find still has a bottom margin.

![image](https://github.com/user-attachments/assets/7c1d801c-2060-4848-91cf-20dd36aa0122)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/caacfcab86d98070b1662e661c8f65661b8198b9

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+		background: none;
+		padding: 0 $gap-smaller;
+		margin: 0;
+		display: inline-flex;
```
The selector priority of this style is higher than `.wp-core-ui .button.media-frame-menu-toggle`, which leads to a hidden "Menu" button being shown after clicking the "Add product gallery images" button within Product gallery.

![image](https://github.com/user-attachments/assets/5870dcd1-f8b1-4831-ae11-b09cadbd1e67)

tomalec replied:
I'll address it after extracting rules to a separate file, as the order and specificity may be changed there

tomalec replied:
This is the biggest downside of scoping the changes to the product editor page only with `.post-php.post-type-product` - it significantly increases the specificity of all selectors. Introducing the problems like this one.

I can scope out this particular instance, but it's hard to predict and prevent all the other cases.

tomalec replied:
Alternatively, I could not set the `display` property and leave it as it was styling it as `inline-block` (set by wp core).
It will help to prevent drastic failures like this one when the hidden element is shown.

However, the margin-based styling of internals is more flaky IMO, plus it will not prevent the other styles being overwritten for the rules with specificity lower than 0-3-0

tomalec replied:
I removed `display: flex` in https://github.com/woocommerce/woocommerce/pull/52632/commits/4ce8050217b223d0b10de88c192cb82d300706db so it should solve at least the problem of non-hideable elements. 

However, it has a little downside, as the whitespace in inline items does not fully collapse (to 0 widths), which makes the gap between the media icon and the text one space wider:
![image](https://github.com/user-attachments/assets/2b48658b-f66a-47ab-a5f8-ce82cf276bec)

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+	#screen-meta-links .show-settings {
+		background: none;
+		padding: 0 $gap-smaller;
+		margin: 0;
```
The top margin of the "Add to gallery" button is overridden. The button can be found after clicking the "Add product gallery images" button within Product gallery.

Before:
![image](https://github.com/user-attachments/assets/a73b836f-713c-4fd8-b4b7-c71ee3968370)

After:
![image](https://github.com/user-attachments/assets/870051de-ef48-4328-8489-361691ebbb8c)

tomalec replied:
Fixed in 257f2e83609ac55ac96f714f2394342dbdde599c

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+	// Link color.
+	a,
+	.button-link  {
+		color:  var(--wp-admin-theme-color, $studio-wordpress-blue);
```
Not sure if it's expected but the text color of the "Delete permanently" button is overridden. The button can be found after clicking the product image.

Before:
![image](https://github.com/user-attachments/assets/77294242-7c2e-4e6f-88d6-dea70ed6d8f8)

After:
![image](https://github.com/user-attachments/assets/ac6bd90e-7053-4d86-9881-4f78df19b543)

The same thing happens with the “Move to Trash” button on product and coupon pages.

tomalec replied:
I'm thinking of removing those rules altogether and leaving it up to a separate project to update all the link colors as part of the Woo 2.0 colors update (p7bje6-6wQ-p2).

Having this overly specific selector creates many problems as this one. Also, having it here for the sake of setting it to `wp-admin-theme-color`, which is set to an older color anyway, does not seem worth it.

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/15ce0887208aa5b0fe74d332c51d0a03b175d514

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
+				margin: $gap-smallest;
+			}
+		}
+		.add_product_images{
```
```suggestion
		.add_product_images {
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -5801,13 +6290,92 @@ img.help_tip {
-				background-color: #eee;
+				color: var(--Alias-text-text, $alias-text);
+				font-weight: 700;
+				border-left: 3px solid var(--wp-admin-theme-color, $studio-wordpress-blue);
```
How about setting the left border to 3px transparent in the following selector? That way, the tab items will be better aligned.

https://github.com/woocommerce/woocommerce/blob/daf105933d5bb378bcecb75afff70bd36ef51c33/plugins/woocommerce/client/legacy/css/admin.scss#L6354

![1](https://github.com/user-attachments/assets/e8b204c0-a04f-4ab8-b04b-a4783a1d2bef)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/8b022a28d2a94323c78e436aa566793e39bdf668

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -7529,40 +8097,55 @@ table.bar_chart {
+	#woocommerce-product-data,
+	#woocommerce-coupon-data {
+		.options_group,
+		// General coupos data does not have options group wrapper, make it look like it has.
```
Typo: coupos

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/153bf863da0e1e18882872f97cc7744ca964bcc5

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -7529,40 +8097,55 @@ table.bar_chart {
+	#woocommerce-coupon-data {
+		.options_group,
+		// General coupos data does not have options group wrapper, make it look like it has.
+		&#general_coupon_data {
```
The compiled selector `#woocommerce-coupon-data#general_coupon_data` seems to be incorrect.

![image](https://github.com/user-attachments/assets/e20ca80b-a31d-4d24-a62f-2364c3b0692c)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/150f2be35a55eafcbe60721ba51315e72b5e90c4

eason9487 commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -7529,40 +8097,55 @@ table.bar_chart {
-		width: 90%;
+		.options_group,
+		// General coupos data does not have options group wrapper, make it look like it has.
+		&#general_coupon_data {
```
The compiled selector `#woocommerce-coupon-data#general_coupon_data` seems to be incorrect.

![image](https://github.com/user-attachments/assets/cf3e3675-e371-46f0-8af0-e2013d65350f)

tomalec replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/52632/commits/150f2be35a55eafcbe60721ba51315e72b5e90c4

eason9487 reviewed (commented):
Thanks for the work. Left some feedback and a few issues/questions.

tomalec committed:
Apply suggestions from code review

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Remove  not needed selector

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1858261992

tomalec committed:
Fix vertical alignment ofr wp-media button icon

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860104575

tomalec committed:
Remove unused style

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860124472

tomalec committed:
Shift input placeholder label by 1px

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860172092

tomalec committed:
Fix slug color

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860178353

tomalec committed:
Remove redundant rules

tomalec committed:
Overwrite marketplace suggestion button styles

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860300000

tomalec committed:
Fix selector for `the-tagcloud`

Address https://github.com/woocommerce/woocommerce/pull/52632/files/daf105933d5bb378bcecb75afff70bd36ef51c33#r1860395444

tomalec committed:
Remove link color

to avoid overspecificity of selectors.
Leave it up to a separate project to update them all.

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860419973

tomalec committed:
Align tabs vertically

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860479395

tomalec committed:
Fix typo in code comment

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860496917

tomalec committed:
Fix the selector for `#general_coupon_data`

Address
https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860499233
https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860500108

tomalec committed:
Fix sizing of category tabs

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860354302

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

tomalec committed:
Extract new Product editor styles to a separate file

tomalec committed:
Move remaining product-data styles to the separate file

tomalec committed:
Remove dead code,

there is no trace of `dp-applied` in this the current version of WooCommerce.
The line originates from the very first initial commit from Jigoshop source

https://github.com/woocommerce/woocommerce/commit/8e8613b116caed4ce13dd0610deddcc3e3df3992

tomalec committed:
Fix some stylelint errors

tomalec committed:
Merge some rulesets

tomalec committed:
Consolidate `#title` styles

tomalec commented:
To maintain compatibility with WooCommerce Gift Cards extension, I'll leave the color of the border between option groups as it was `#eee` instead of `#ededed`. IMO, the colors are relatively similar, and by not overwriting it, we won't increase the specificity of the selector, and Gift Cards trick will still work:
![image](https://github.com/user-attachments/assets/aac45683-fbe4-4b03-acf4-e957983497f1)

tomalec committed:
Fix Gift Cards compatibility

tomalec commented:
#### Known compat quirks

##### WooCommerce Gift Cards

We introduced padding for option groups. WooCommerce Gift Cards creates and option group with a header inside, which is styled to look like being over the option group separators.

![image](https://github.com/user-attachments/assets/8b3c5ed7-6aa4-4459-9035-e76f574b94e2)

With padding added to option groups, the header is offset lower to the bottom:

![image](https://github.com/user-attachments/assets/eecc678e-58db-4ff5-b1e0-98cf144def55)

The following rules could be added to WooCommerce Gift Cards to fix that problem:
```css
#giftcard_settings {
	padding-top: 0;
}
#giftcard_settings .hr-section {
   margin-bottom: 36px;
}
```

tomalec committed:
Restore margin-based gaps

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860323686

tomalec committed:
Implement gaps & gutters using margins

and border instead of flex containers to avoid "non-hdeable" elements due to too high specificity.

tomalec committed:
Use dedicated variable for postbox padding.

tomalec committed:
Improve support for extensions adding category tabs

tomalec committed:
Revert buttons display to whatever it was

to avoid problems with hiding

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860408726

tomalec committed:
Fix margin for sold individualy field

Missed in f4d0f453c97798d9ae9ab872c5a80491b0ef0562

tomalec committed:
Revert unification of buttons' margins

As the selector has too high specificity, and would overwrite all customizations.

Address https://github.com/woocommerce/woocommerce/pull/52632#discussion_r1860414861

tomalec committed:
Fix gaps between postboxes

tomalec committed:
Improve description editor look like postbox in no-js env

tomalec committed:
Reduce FOUC before `woocommerce-product-description` class is attached

tomalec committed:
Reduce specificity of some selectors.

tomalec committed:
Remove prefixes from placeholder selectors

tomalec committed:
Fix some linting errors

tomalec committed:
Reuse breakpoint variables

tomalec committed:
Import product editor styles at the bottom

to overrride wc-tabs with media queries.

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

tomalec commented:
Thank you @eason9487 for the great comments.

I hope I have addressed them all. Sorry that I made quite a big change to the code base after your initial review. But I hope having the code separated will make it easier to work with in the future.

After addressing your comments directly in the old codebase (commit references in the answers above)
I made a number of other changes:
- Extracted product editor styles to a separate file
- Reverted flexbox layout and gaps to a margin-based solution for 
   - options panel
   - postbox insides
- Introduced a dedicated variable for padding, in case we want to make it different from the regular gap, and to make calculations more conceptually explicit
- Tweaked selector for category tabs, so it also covers other extensions, like brands or coupons (https://github.com/woocommerce/woocommerce/pull/52632/commits/504c312c1549aa3f852923b5f28cd9e3a9df062f)
- Made the description editor in no-JS mode look more like other panels (https://github.com/woocommerce/woocommerce/pull/52632/commits/a1e47261128881a68c13aa9b1bc3a3c031ebf458)
- Reduce the FOUC for rich text editor (https://github.com/woocommerce/woocommerce/pull/52632/commits/477bf15668782f8dde82875d67deb6c2f4e6d446)  - this could be somewhat separated from the editor, but I think it's worth piggybacking.
- Reduced the specificity of used selectors - this may make it harder to "broaden" the styles in the future from just the editor page, to other pages as well. But given how much I struggled with already to high specificity of some rules, I'd rather keep it lower (https://github.com/woocommerce/woocommerce/pull/52632/commits/b5d274861729638d6d8f92935bf8b0562573cbae)
- Removed prefixes from placeholder rules (https://github.com/woocommerce/woocommerce/pull/52632/commits/9c4e861e3dd2294eea2ec881fffa965249046d9c)
- Fix some linting rules.

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

eason9487 commented on plugins/woocommerce/client/legacy/css/admin/product-editor.scss:
```diff
@@ -0,0 +1,748 @@
+			margin: $gap calc(48px - 20px) $gap calc(48px - 12px);
+
+			@media only screen and (max-width: $break-wide) {
+				margin: $gap calc(gap-larger - 20px) $gap calc($gap-larger - 12px);
```
Variable is missing the starting `$`.

```suggestion
				margin: $gap calc($gap-larger - 20px) $gap calc($gap-larger - 12px);
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin/product-editor.scss:
```diff
@@ -0,0 +1,748 @@
+		fieldset.form-field,
+		.downloadable_files {
+			// Use margin as gap and horizontal padding.
+			// Accomodate for p trick with label that uses padding as margin
```
Is "Accomodate" a typo?

```suggestion
			// Accommodate for p trick with label that uses padding as margin
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin/product-editor.scss:
```diff
@@ -0,0 +1,748 @@
+		 */
+		&#general_coupon_data {
+			border-bottom-color: #ededed;
+			// Deduct vertival margins of inner paragraphs.
```
```suggestion
			// Deduct vertical margins of inner paragraphs.
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin/product-editor.scss:
```diff
@@ -0,0 +1,748 @@
+/**
+ * product-editor.scss
+ * Styles for refubrished product and coupon editor.
```
```suggestion
 * Styles for refurbished product and coupon editor.
```

eason9487 reviewed (commented):
Thanks for the work! The change looks good.

There are a few typos and a Sass variable that is not applied correctly.

Besides, I've tried to find every possible place but couldn't find it anywhere. Could you point out where these CSS variables are defined?

- `--Alias-text-text`
- `--Alias-text-text-tertiary`
- `--Gray-Gray-30`
- `--Gray-Gray-60`
- `--grid-unit-50`
- `--Gutenberg-White`

---

Note that there used to be an additional 10px between the header and the "Add new product" or "Edit product" title, but [another PR](https://github.com/woocommerce/woocommerce/pull/52610/files#diff-02a7ac5265a00491ebc0bd6aa4218770f380088d62c3263c5a18e2396b8bdb8eR9-R11) recently added a new style overriding that. It makes the title look a bit incongruous.

<table>
<tr><th>Before</th><th>After</th><th>After + without the overriding</th></tr>
<tr><td>

![image](https://github.com/user-attachments/assets/3b46fb76-ced6-4d2e-822e-d8270f3eb4ab)

</td><td>

![image](https://github.com/user-attachments/assets/54db6cf9-3b39-4e6a-8699-e35a36f2eb78)

</td><td>

![image](https://github.com/user-attachments/assets/fbc699fa-40bc-4a32-b7ba-6cdfaf85803a)

</td></tr></table>

I think perhaps it's not appropriate for this PR to override that style again, and the repeated title will probably be removed eventually, but it's still worth documenting the concern.

tomalec committed:
Apply suggestions from code review

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec commented:
Thanks for the review :) 

> Besides, I've tried to find every possible place but couldn't find it anywhere. Could you point out where these CSS variables are defined?

I found those in Figma document, and noticed, there are some areas in WC codebase that already use those, like 

https://github.com/woocommerce/woocommerce/blob/fe2de3a170fd09ff96d0e68820a4d119622d0a99/plugins/woocommerce/client/admin/client/settings-payments/components/status-badge/status-badge.scss#L18

https://github.com/woocommerce/woocommerce/blob/6d3956ae38c452dbecc61b6d4df11d0ac722066d/plugins/woocommerce/client/admin/client/marketplace/components/promo-card/promo-card.scss#L114

So I assumed there is a convention of using them that way, maybe with a premise to eventually introduce them somewhere or at least indicate where the value comes from a common grid system or is given more specifically/custom.

tomalec committed:
Merge branch 'trunk' into update/product-editor-css

eason9487 reviewed (approved):
Thank you!

I found the uses of similar CSS variables in this codebase and some repos as well but never found any variable definitions. From the description of the PRs that introduced them, I didn't find any clues as to how they are used either.

It looks more like they were accidentally ported from Figma since these CSS variables are not widely applied to the most generic components and layouts. For example, imagine what the scenario would be like to make the color of the status badges customizable only on the Payment Settings page.

Therefore, I lean toward removing these CSS variables that seem to only be effective in Figma. In any case, I believe it won't need another round of code reviews even if there are further tweaks related to this.

tomalec committed:
Remove unused CSS variables

Address https://github.com/woocommerce/woocommerce/pull/52632#pullrequestreview-2497500363

tomalec committed:
Merge branch 'trunk' into update/product-editor-css
